internal/eval/eval.go

```diff
@@ -235,7 +235,7 @@ func evalMSET(args []string, store *dstore.Store) []byte {
 	insertMap := make(map[string]*object.Obj, len(args)/2)
 	for i := 0; i < len(args); i += 2 {
 		key, value := args[i], args[i+1]
-		oType, oEnc := deduceTypeEncoding(value)
+		oType, oEnc := object.ObjTypeString, object.ObjEncodingRaw
```

Comment:
I believe the fix should be the save the val by encoding it into the associated type similar to `evalSET` operation.
Redis does the same thing trying to deduce the encoding type and save it accordingly.
```
for (j = 1; j < c->argc; j += 2) {
        c->argv[j+1] = tryObjectEncoding(c->argv[j+1]);
        setKey(c, c->db, c->argv[j], c->argv[j + 1], setkey_flags);
```

Reply:
@JyotinderSingh please correct if I'm missing something here.

Reply:
That's correct. We should be using the exact same type deduction logic for both set and mset